qiskit_experiments/framework/composite/composite_experiment_data.py

```diff
@@ -103,3 +98,82 @@ def _add_single_data(self, data):
                     else:
                         sub_data["counts"] = data["counts"]
                 self._components[index]._add_single_data(sub_data)
+
+    def save(self, verbose=True) -> None:
+        super().save(verbose)
+        for comp in self._components:
+            comp.save(verbose=False)
+
+    def save_metadata(self) -> None:
+        super().save_metadata()
+        for comp in self._components:
+            comp.save_metadata()
+
+    @classmethod
+    def load(cls, experiment_id: str, service: DatabaseServiceV1) -> "CompositeExperimentData":
+        expdata = DbExperimentDataV1.load(experiment_id, service)
+        components = []
+        for comp_id, comp_class in zip(
+            expdata.metadata["component_ids"], expdata.metadata["component_classes"]
+        ):
+            load_class = globals()[comp_class]
+            load_func = getattr(load_class, "load")
+            loaded_comp = load_func(comp_id, service)
+            components.append(loaded_comp)
+
+        expdata.__class__ = CompositeExperimentData
```

Comment:
This is a little sketchy and what I tried to avoid in #423. But currently composite experiment data has an issue where it requires `experiments` to not be None on initialization, because it gets components from there.

I think it might be better to modify the `init` method of htis class to make experiments optional, and to also add an optional `components` kwarg that you can directly pass in the component experiment data during init.

Reply:
But, even when `CompositeExperimentData` accepts `components`, we will not use it and not call it from `CompositeExperimentData.load`. Because `expdata` was already initialized in `DbExperimentData.load`.